packages/table-core/src/columnHelper.ts

```diff
@@ -49,22 +49,24 @@ import { DeepKeys, DeepValue, RequiredKeys } from './utils'
 //   cell: info => info.getValue(),
 // })
 
-export type ColumnHelper<TData extends RowData> = {
-  accessor: <
-    TAccessor extends AccessorFn<TData> | DeepKeys<TData>,
-    TValue extends TAccessor extends AccessorFn<TData, infer TReturn>
-      ? TReturn
-      : TAccessor extends DeepKeys<TData>
+type Accessor<TData extends RowData> = <
+  TAccessor extends AccessorFn<TData> | string,
```

Comment:
`DeepKeys<TData>` is expensive, and the result of calling it doesn't seem to be cached by the compiler. We can replace the first type check with `string`, then, if the accessor is a function, we can short circuit and avoid ever calling `DeepKeys`. If the accessor is not a function, we simply narrow the type from `string` to a key.